Lib/python/pyrun.swg

```diff
@@ -279,18 +283,26 @@ SwigPyClientData_New(PyObject* obj)
     /* the newraw method and newargs arguments used to create a new raw instance */
     if (PyClass_Check(obj)) {
       data->newraw = 0;
-      data->newargs = obj;
       Py_INCREF(obj);
+      data->newargs = obj;
     } else {
       data->newraw = PyObject_GetAttrString(data->klass, "__new__");
       if (data->newraw) {
-	Py_INCREF(data->newraw);
-	data->newargs = PyTuple_New(1);
-	PyTuple_SetItem(data->newargs, 0, obj);
+        data->newargs = PyTuple_New(1);
+        if (data->newargs) {
+          Py_INCREF(obj);
+          PyTuple_SET_ITEM(data->newargs, 0, obj);
+        } else {
+          Py_DECREF(data->newraw);
```

Comment:
I considered the options of:

1. Setting `data->destroy = 0` here and then just invoking `SwigPyClientData_Del()`
2. Having cleanup code at the end of the function and using `goto fail;`, etc.
3. Putting the cleanup inline (as I have done)

If you feel strongly that one of the other options would be preferable, I am happy to update accordingly.

Reply:
The cleanup is simple and fine to leave here.